tests/integration.rs

```diff
@@ -6433,6 +6434,7 @@ mod common_parallel {
     }
 
     #[test]
+    #[cfg(not(feature = "mshv"))]
```

Comment:
What bit of vfio-user doesn't work? It's not the same as VFIO.

Reply:
This is a good catch. I would think the test should be fine as long as the nvme backend (from SPDK) works fine.

Reply:
Looking into it. Keep you updated.

Reply:
The first failure is here:
        assert!(exec_host_command_status(
                "/usr/local/bin/spdk-nvme/rpc.py nvmf_create_subsystem nqn.2019-07.io.spdk:cnode -a -s test"
            )
            .success());

Error while connecting to /var/tmp/spdk.sock
Is SPDK application running?
Error details: Invalid or non-existing address: '/var/tmp/spdk.sock'
thread 'common_parallel::test_vfio_user' panicked at 'assertion failed: exec_host_command_status(\"/usr/local/bin/spdk-nvme/rpc.py nvmf_create_transport -t VFIOUSER\").success()', tests/integration.rs:6407:9

Reply:
Does the host need any configuration?

Reply:
No additional configurations needed besides what's done from `setup_spdk_nvme()`: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/tests/integration.rs#L6381.

I would suggest you to check if spdk-nvme works on your host by itself (independent of Cloud Hypervisor). Here is the version of SPDK we used and how it was compiled: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/resources/Dockerfile#L115. Also you can get some references to the spdk-nvme repo from this document: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/vfio-user.md#example-nvme-device.

Reply:
@likebreath  Looks like the test is also failing on KVM. Similar error.

Reply:
I tried to run manually following command and saw a core dump
# /usr/local/bin/spdk-nvme/nvmf_tgt -i 0 -m 0x1
Illegal instruction (core dumped)
#

Reply:
Do you see anything in dmesg?

Reply:
I can see this line:
[ 4394.865034] traps: nvmf_tgt[16976] trap invalid opcode ip:64cb7b044290 sp:7fff2af2afa0 error:0 in nvmf_tgt[64cb7afbe000+1ab000]

Reply:
Is there anything else after that line? Does it show the invalid opcode?

Reply:
No there is nothing after that.

Reply:
> No additional configurations needed besides what's done from `setup_spdk_nvme()`: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/tests/integration.rs#L6381.
> 
> I would suggest you to check if spdk-nvme works on your host by itself (independent of Cloud Hypervisor). Here is the version of SPDK we used and how it was compiled: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/resources/Dockerfile#L115. Also you can get some references to the spdk-nvme repo from this document: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/vfio-user.md#example-nvme-device.

I tried those steps to configure vfio user device and pass it to CLH. I got this error:

thread 'vcpu0' panicked at 'Failed cloning interrupt's EventFd: Os { code: 24, kind: Uncategorized, message: "Too many open files" }', vmm/src/interrupt.rs:82:18

Reply:
This is promising. I remember the SPDK-NVMe backend did open quite many files. Can you try to increase the ulimit value for open files? You may need to set both the softlimit (`ulimit -n`) and the hard limit (via `/etc/security/limits.conf`).

Also, I guess the Illegal Instruction issue was gone after you manually compiled the SPDK repo?

Reply:
@likebreath @liuw I have tested the vfio_uset test case manually after
modifying some configurations of of the Bare-metal Host. The test case passes on mshv.
Looks like in my host  ghcr.io/cloud-hypervisor/cloud-hypervisor image  has issues.